src/sentry/workflow_engine/processors/delayed_workflow.py

```diff
@@ -483,27 +517,26 @@ def get_groups_to_fire(
 
     for dcg in data_condition_groups:
         slow_conditions = dcg_to_slow_conditions[dcg.id]
-        action_match = DataConditionGroup.Type(dcg.logic_type)
         workflow_id = event_data.dcg_to_workflow.get(dcg.id)
         workflow_env = workflows_to_envs[workflow_id] if workflow_id else None
 
         for group_id in event_data.dcg_to_groups[dcg.id]:
-            conditions_to_evaluate: list[tuple[DataCondition, list[int | float]]] = []
-            for condition in slow_conditions:
-                unique_queries = generate_unique_queries(condition, workflow_env)
-                query_values = [
-                    condition_group_results[unique_query][group_id]
-                    for unique_query in unique_queries
-                ]
-                conditions_to_evaluate.append((condition, query_values))
-
-            evaluation = evaluate_data_conditions(conditions_to_evaluate, action_match)
-            if (
-                evaluation.logic_result and workflow_id is None
-            ):  # TODO: detector trigger passes. do something like create issue
-                pass
-            elif evaluation.logic_result:
-                groups_to_fire[group_id].add(dcg)
+            try:
+                result = _group_result_for_dcg(
+                    group_id, dcg, workflow_env, condition_group_results, slow_conditions
+                )
+                if (
+                    result and workflow_id is None
+                ):  # TODO: detector trigger passes. do something like create issue
+                    pass
+                elif result:
+                    groups_to_fire[group_id].add(dcg)
+            except MissingQueryResult:
+                # If we didn't get complete query results, don't fire.
+                metrics.incr("workflow_engine.delayed_workflow.missing_query_result")
+                logger.warning(
+                    "workflow_engine.delayed_workflow.missing_query_result", exc_info=True
```

Comment:
Will `UniqueConditionQuery` be splatted in a readable manner into the logs? It's a `dataclass` and has a `__repr__` but I'm not sure how/if `exc_info` uses it